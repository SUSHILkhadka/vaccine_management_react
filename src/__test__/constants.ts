import { IAllergy } from '../interface/IAllergy';
import { IVaccine } from '../interface/IVaccine';

export const vaccine: IVaccine = {
  id: 24,
  name: 'FluMist',
  description: 'Used in Seasonal Flu. Its other name is LAIV.',
  numberOfDoses: 2,
  releaseDate: '2008-01-30',
  photoUrl: '',
  isMandatory: false,
};

export const allergyList: IAllergy[] = [
  {
    id: -1,
    name: 'allery1',
    status: 'added',
  },
  {
    id: 3,
    name: 'allery3',
    status: 'added',
  },
  {
    id: 3,
    name: 'allery3',
    status: 'deleted',
  },
];

export const vaccineArray = [
  {
    id: 1,
    name: 'a',
    description: '',
    numberOfDoses: 1,
    releaseDate: new Date().toLocaleDateString(),
    photoUrl: '',
    isMandatory: false,
  },
  {
    id: 2,
    name: 'vaccinenameiswired',
    description: '',
    numberOfDoses: 1,
    releaseDate: new Date().toLocaleDateString(),
    photoUrl: '',
    isMandatory: true,
  },
  {
    id: 3,
    name: 'a',
    description: '',
    numberOfDoses: 1,
    releaseDate: new Date().toLocaleDateString(),
    photoUrl: '',
    isMandatory: false,
  },
  {
    id: 4,
    name: 'a',
    description: '',
    numberOfDoses: 1,
    releaseDate: new Date().toLocaleDateString(),
    photoUrl: '',
    isMandatory: true,
  },
];
