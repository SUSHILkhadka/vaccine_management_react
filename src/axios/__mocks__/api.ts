import { vaccineArray } from '../../__test__/utils/sort.test';

module.exports = {
  post: (url: string, body: any = "without any body") => {
    return {
      data: {
        data: body,
      },
    };
  },
  get: (url: string) => {
    return {
      data: {
        data: vaccineArray,
      },
    };
  },
  put: (url: string, body: any) => {
    return {
      data: {
        data: body,
      },
    };
  },
  delete: (url: string, body: any = "without any body") => {
    return {
      data: {
        data: body,
      },
    };
  },
};
