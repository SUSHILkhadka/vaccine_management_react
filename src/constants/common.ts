// export const URL_TO_BACKEND = 'http://localhost:8000';
export const URL_TO_BACKEND = 'https://fierce-peak-73671.herokuapp.com';
// export const imageURL = 'https://www.gravatar.com/avatar/2c7d99fe281ecd3bcd65ab915bac6dd5?s=250';
export const imageURL =
  'https://images.theconversation.com/files/411706/original/file-20210716-21-lunh2y.jpg?ixlib=rb-1.1.0&q=45&auto=format&w=1200&h=1200.0&fit=crop';
export const IMAGE_COMPRESSION_PERCENTAGE = 100;
export const FIXED_FILE_SIZE = 200 * 1000; //around 200kb
export const IMAGE_LOGO1 = '../../assets/logo1.png';
export const IMAGE_LOGO2 = '../../assets/logo2.png';
