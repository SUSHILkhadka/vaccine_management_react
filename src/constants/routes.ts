export const PATH_ROOT = '/';
export const PATH_LOGIN = '/login';
export const PATH_REGISTER = '/register';
export const PATH_SETTING = '/setting';
export const PATH_VACCINE_ADD = '/vaccine/add';
export const PATH_VACCINE_TABLE = '/';
export const PATH_VACCINE_EDIT = '/vaccine/edit';
