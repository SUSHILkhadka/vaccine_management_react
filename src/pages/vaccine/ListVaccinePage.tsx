import { Skeleton } from 'antd';
import React, { useEffect, useState } from 'react';
import { readAllVaccines } from '../../axios/backendVaccine';
import CustomSort from '../../components/Customs/CSort/CSort';
import CTable from '../../components/Customs/CTable/CTable';
import { IVaccine } from '../../interface/IVaccine';
import { errorMessage, showDefaultErrorMessage } from '../../utils/message';

const ListVaccinePage: React.FC = () => {
  const [loading, setLoading] = useState<boolean>(true);
  const [dataOriginal, setDataOrignal] = useState<IVaccine[]>([]);
  const [dataToDisplay, setDataToDisplay] = useState<IVaccine[]>([]);
  const [reload, setReload] = useState<boolean>(false);

  const handleReload = () => {
    setReload((current) => !current);
  };

  useEffect(() => {
    let isMounted = true;

    const getalldata = async () => {
      try {
        const vaccines = await readAllVaccines();
        if (isMounted) {
          setDataOrignal(vaccines.data);
          setDataToDisplay(vaccines.data);
        }
      } catch (e: any) {
        if (e.response && e.response.data) errorMessage(e.response.data.message);
        else showDefaultErrorMessage();
        setDataToDisplay([]);
      }
      setLoading(false);
    };
    getalldata();

    return () => {
      isMounted = false;
    };
  }, [reload]);

  return (
    <div className='page--listvaccine'>
        <CustomSort
          dataOriginal={dataOriginal}
          setDataToDisplay={setDataToDisplay}
        />
      {loading ? (
        <Skeleton active />
      ) : (
        <CTable Obj={dataToDisplay} reloadHandler={handleReload} />
      )}
    </div>
  );
};
export default ListVaccinePage;
