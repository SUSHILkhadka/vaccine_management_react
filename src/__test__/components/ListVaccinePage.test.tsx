import { act } from 'react-dom/test-utils';
import ListVaccinePage from '../../pages/vaccine/ListVaccinePage';
import '../../__mocks__/matchMedia';
import { allergyList, vaccine } from '../constants';
import { renderWithProvidersAndNavigation } from './customRender';

jest.mock('../../axios/api');
it('should render list vaccine page properly', async () => {
  const page = renderWithProvidersAndNavigation(<ListVaccinePage />, {
    preloadedState: {
      vaccine: vaccine,
      allergy: allergyList,
    },
  });

  await act(async () => {
    page.rerender(<ListVaccinePage />);
  });
  expect(page.asFragment()).toMatchSnapshot();

});
