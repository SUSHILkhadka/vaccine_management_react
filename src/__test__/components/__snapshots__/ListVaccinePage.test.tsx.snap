// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render list vaccine page properly 1`] = `
<DocumentFragment>
  <div
    class="page--listvaccine"
  >
    <div
      class="custom--dropdown--container"
    >
      <div
        class="custom--dropdown--button"
      >
        <div>
          <span
            class="custom--dropdown--span"
          >
            âœ•
          </span>
          Show Favourites Only
        </div>
      </div>
    </div>
    <div
      class="table--container"
    >
      <div
        class="ant-table-wrapper actual--table"
      >
        <div
          class="ant-spin-nested-loading"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table"
              data-testid="table"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup>
                      <col
                        style="width: 10%;"
                      />
                      <col
                        style="width: 20%;"
                      />
                      <col
                        style="width: 20%;"
                      />
                      <col
                        style="width: 20%;"
                      />
                    </colgroup>
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          Photograph
                        </th>
                        <th
                          aria-sort="ascending"
                          class="ant-table-cell ant-table-column-sort ant-table-column-has-sorters"
                          style="text-align: center;"
                          tabindex="0"
                        >
                          <div
                            class="ant-table-filter-column"
                          >
                            <span
                              class="ant-table-column-title"
                            >
                              <div
                                class="ant-table-column-sorters"
                              >
                                <span
                                  class="ant-table-column-title"
                                >
                                  Name
                                </span>
                                <span
                                  class="ant-table-column-sorter ant-table-column-sorter-full"
                                >
                                  <span
                                    class="ant-table-column-sorter-inner"
                                  >
                                    <span
                                      aria-label="caret-up"
                                      class="anticon anticon-caret-up ant-table-column-sorter-up active"
                                      role="presentation"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        data-icon="caret-up"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="0 0 1024 1024"
                                        width="1em"
                                      >
                                        <path
                                          d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                        />
                                      </svg>
                                    </span>
                                    <span
                                      aria-label="caret-down"
                                      class="anticon anticon-caret-down ant-table-column-sorter-down"
                                      role="presentation"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        data-icon="caret-down"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="0 0 1024 1024"
                                        width="1em"
                                      >
                                        <path
                                          d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                        />
                                      </svg>
                                    </span>
                                  </span>
                                </span>
                              </div>
                            </span>
                            <span
                              class="ant-dropdown-trigger ant-table-filter-trigger"
                              role="button"
                              tabindex="-1"
                            >
                              <span
                                aria-label="search"
                                class="anticon anticon-search"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="search"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </div>
                        </th>
                        <th
                          aria-sort="ascending"
                          class="ant-table-cell ant-table-column-sort ant-table-column-has-sorters"
                          style="text-align: center;"
                          tabindex="0"
                        >
                          <div
                            class="ant-table-column-sorters"
                          >
                            <span
                              class="ant-table-column-title"
                            >
                              Mandatory
                            </span>
                            <span
                              class="ant-table-column-sorter ant-table-column-sorter-full"
                            >
                              <span
                                class="ant-table-column-sorter-inner"
                              >
                                <span
                                  aria-label="caret-up"
                                  class="anticon anticon-caret-up ant-table-column-sorter-up active"
                                  role="presentation"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="caret-up"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="0 0 1024 1024"
                                    width="1em"
                                  >
                                    <path
                                      d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                    />
                                  </svg>
                                </span>
                                <span
                                  aria-label="caret-down"
                                  class="anticon anticon-caret-down ant-table-column-sorter-down"
                                  role="presentation"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="caret-down"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="0 0 1024 1024"
                                    width="1em"
                                  >
                                    <path
                                      d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </span>
                          </div>
                        </th>
                        <th
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          Action
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="4"
                      >
                        <td
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          <img
                            alt="Loading"
                            class="img-avatar-table"
                            src="https://images.theconversation.com/files/411706/original/file-20210716-21-lunh2y.jpg?ixlib=rb-1.1.0&q=45&auto=format&w=1200&h=1200.0&fit=crop"
                          />
                        </td>
                        <td
                          class="ant-table-cell ant-table-column-sort"
                          style="text-align: center;"
                        >
                          a
                        </td>
                        <td
                          class="ant-table-cell ant-table-column-sort"
                          style="text-align: center;"
                        >
                          <div
                            class="table-favourite"
                          >
                            <span
                              aria-label="star"
                              class="anticon anticon-star"
                              role="img"
                              style="color: firebrick;"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="star"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                                />
                              </svg>
                            </span>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          <div
                            class="ant-space ant-space-horizontal ant-space-align-center"
                          >
                            <div
                              class="ant-space-item"
                              style="margin-right: 16px;"
                            >
                              <button
                                class="ant-btn ant-btn-default button button--edit"
                                type="button"
                              >
                                <span>
                                  Edit
                                </span>
                              </button>
                            </div>
                            <div
                              class="ant-space-item"
                            >
                              <button
                                class="ant-btn ant-btn-default button button--delete"
                                type="button"
                              >
                                <span>
                                  Delete
                                </span>
                              </button>
                            </div>
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="2"
                      >
                        <td
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          <img
                            alt="Loading"
                            class="img-avatar-table"
                            src="https://images.theconversation.com/files/411706/original/file-20210716-21-lunh2y.jpg?ixlib=rb-1.1.0&q=45&auto=format&w=1200&h=1200.0&fit=crop"
                          />
                        </td>
                        <td
                          class="ant-table-cell ant-table-column-sort"
                          style="text-align: center;"
                        >
                          vaccinenameiswired
                        </td>
                        <td
                          class="ant-table-cell ant-table-column-sort"
                          style="text-align: center;"
                        >
                          <div
                            class="table-favourite"
                          >
                            <span
                              aria-label="star"
                              class="anticon anticon-star"
                              role="img"
                              style="color: firebrick;"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="star"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                                />
                              </svg>
                            </span>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          <div
                            class="ant-space ant-space-horizontal ant-space-align-center"
                          >
                            <div
                              class="ant-space-item"
                              style="margin-right: 16px;"
                            >
                              <button
                                class="ant-btn ant-btn-default button button--edit"
                                type="button"
                              >
                                <span>
                                  Edit
                                </span>
                              </button>
                            </div>
                            <div
                              class="ant-space-item"
                            >
                              <button
                                class="ant-btn ant-btn-default button button--delete"
                                type="button"
                              >
                                <span>
                                  Delete
                                </span>
                              </button>
                            </div>
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="1"
                      >
                        <td
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          <img
                            alt="Loading"
                            class="img-avatar-table"
                            src="https://images.theconversation.com/files/411706/original/file-20210716-21-lunh2y.jpg?ixlib=rb-1.1.0&q=45&auto=format&w=1200&h=1200.0&fit=crop"
                          />
                        </td>
                        <td
                          class="ant-table-cell ant-table-column-sort"
                          style="text-align: center;"
                        >
                          a
                        </td>
                        <td
                          class="ant-table-cell ant-table-column-sort"
                          style="text-align: center;"
                        >
                          <div
                            class="table-favourite"
                          >
                            <span
                              aria-label="star"
                              class="anticon anticon-star"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="star"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
                                />
                              </svg>
                            </span>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          <div
                            class="ant-space ant-space-horizontal ant-space-align-center"
                          >
                            <div
                              class="ant-space-item"
                              style="margin-right: 16px;"
                            >
                              <button
                                class="ant-btn ant-btn-default button button--edit"
                                type="button"
                              >
                                <span>
                                  Edit
                                </span>
                              </button>
                            </div>
                            <div
                              class="ant-space-item"
                            >
                              <button
                                class="ant-btn ant-btn-default button button--delete"
                                type="button"
                              >
                                <span>
                                  Delete
                                </span>
                              </button>
                            </div>
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="3"
                      >
                        <td
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          <img
                            alt="Loading"
                            class="img-avatar-table"
                            src="https://images.theconversation.com/files/411706/original/file-20210716-21-lunh2y.jpg?ixlib=rb-1.1.0&q=45&auto=format&w=1200&h=1200.0&fit=crop"
                          />
                        </td>
                        <td
                          class="ant-table-cell ant-table-column-sort"
                          style="text-align: center;"
                        >
                          a
                        </td>
                        <td
                          class="ant-table-cell ant-table-column-sort"
                          style="text-align: center;"
                        >
                          <div
                            class="table-favourite"
                          >
                            <span
                              aria-label="star"
                              class="anticon anticon-star"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="star"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
                                />
                              </svg>
                            </span>
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                          style="text-align: center;"
                        >
                          <div
                            class="ant-space ant-space-horizontal ant-space-align-center"
                          >
                            <div
                              class="ant-space-item"
                              style="margin-right: 16px;"
                            >
                              <button
                                class="ant-btn ant-btn-default button button--edit"
                                type="button"
                              >
                                <span>
                                  Edit
                                </span>
                              </button>
                            </div>
                            <div
                              class="ant-space-item"
                            >
                              <button
                                class="ant-btn ant-btn-default button button--delete"
                                type="button"
                              >
                                <span>
                                  Delete
                                </span>
                              </button>
                            </div>
                          </div>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
            <ul
              class="ant-pagination ant-table-pagination ant-table-pagination-right"
              unselectable="unselectable"
            >
              <li
                aria-disabled="true"
                class="ant-pagination-prev ant-pagination-disabled"
                title="Previous Page"
              >
                <button
                  class="ant-pagination-item-link"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="left"
                    class="anticon anticon-left"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="left"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                      />
                    </svg>
                  </span>
                </button>
              </li>
              <li
                class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                tabindex="0"
                title="1"
              >
                <a
                  rel="nofollow"
                >
                  1
                </a>
              </li>
              <li
                aria-disabled="true"
                class="ant-pagination-next ant-pagination-disabled"
                title="Next Page"
              >
                <button
                  class="ant-pagination-item-link"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="right"
                    class="anticon anticon-right"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="right"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                      />
                    </svg>
                  </span>
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
